thirdparty/DiskANN/src/pq_flash_index.cpp

```diff
@@ -1533,30 +1533,21 @@ namespace diskann {
   }
 
   template<typename T>
-  void PQFlashIndex<T>::get_vector_by_ids(const int64_t *ids, const int64_t n, T *output_data) {
-    size_t batch_size = kReadBatchSize;
-    if (long_node) {
-      auto min_size = kReadBatchSize / nsectors_per_node;
-      batch_size = (min_size == 0) ? MAX_N_SECTOR_READS : min_size;
-      if (0 == batch_size) {
-        LOG(ERROR) << "Vector too large, exceeding max number of sector reads";
-        return ;
-      }
-    }
+  void PQFlashIndex<T>::get_vector_by_ids(
+      const std::unordered_map<_u64, std::vector<_u64>> &sectors_to_visit,
+      const int64_t *ids, T *const output_data) {
     ThreadData<T> data = this->thread_data.pop();
     while (data.scratch.sector_scratch == nullptr) {
       this->thread_data.wait_for_push_notify();
       data = this->thread_data.pop();
     }
+
+    const size_t batch_size = std::min(MAX_N_SECTOR_READS / 2UL, sectors_to_visit.size());
+    const size_t half_buf_idx = MAX_N_SECTOR_READS / 2 * read_len_for_node;
```

Comment:
Why we decided to have 2 batch to do pipeline?

Reply:
One for submitted buffer, one for last round results buffer